### PR TITLE
Refactoring into submodules

### PR DESCRIPTION
**Pull request by blegat**

### CODE REVIEW

File: src/StochOptInterface/StochOptInterface.jl [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# TODO This module should be replaced by the package https://github.com/JuliaStochOpt/StochOptInterface.jl
+module StochOptInterface
+
+using DocStringExtensions
```
Comment (frapac) [inline src/StochOptInterface/StochOptInterface.jl]: Should we continue to use DocStringExtensions?

File: src/StructDualDynProg.jl [hunk 2]
```diff
@@ -11,37 +11,14 @@ using StructJuMP
...
-include("stopcrit.jl")
-# Cut Generator
-include("cutgen.jl")
-# Path Sampler
-include("sampler.jl")
-# Solution
-include("solution.jl")
+# TODO This module should be replaced by the package https://github.com/JuliaStochOpt/StochOptInterface.jl
+include("StochOptInterface/StochOptInterface.jl")
+const SOI = StochOptInterface
```
Comment (frapac) [inline src/StructDualDynProg.jl]: I like the similitude with MOI!

File: src/StochOptInterface/StochOptInterface.jl (hunk 1 above)
Comment (blegat) [reply src/StochOptInterface/StochOptInterface.jl]: No we can get rid of it :)

Comment (blegat) [reply src/StochOptInterface/StochOptInterface.jl]: What is the reason to stop using it ?

Comment (frapac) [reply src/StochOptInterface/StochOptInterface.jl]: I am in a minimalistic mood for a few months ...
I tend to believe that the fewer the dependancies, the easier the maintenance is!

**Pull request merged**